src/gtc/gtcpp/gtcpp_codegen.py

```diff
@@ -110,7 +137,15 @@ def visit_BuiltInLiteral(self, builtin: BuiltInLiteral, **kwargs: Any) -> str:
             return "false"
         raise NotImplementedError("Not implemented BuiltInLiteral encountered.")
 
-    Literal = as_mako("static_cast<${dtype}>(${value})")
+    def visit_Literal(
+        self, node: gtcpp.Literal, *, in_data_index: bool = False, **kwargs: Any
+    ) -> str:
+        value = self.visit(node.value, **kwargs)
+        if in_data_index:
+            return value
```

Comment:
Does it really make sense to leverage the `visit_Literal` for the data indices? After all you just take `node.value` as is so you can also do that in the caller (note that `self.visit` is a noop for this case and only matters for `value` being a `BuiltInLiteral`).